src/components/GoodsList/GoodsList.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import './GoodsList.css';
+
+class GoodsList extends React.Component {
+  state = {
+    goods: [...this.props.goods],
+  }
+
+  reverseList = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortByABC = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort(),
+    }));
+  };
+
+  resetToDefault = () => {
+    this.setState(state => ({
+      goods: [...this.props.goods],
+    }))
+  }
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  render() {
+    return(
+      <div>
+        <ul>
+          {this.state.goods.map(good => (
+            <li className="good">
```

Comment:
add key attribute
https://reactjs.org/docs/lists-and-keys.html

Reply:
rly... I just forgot about it) Thanks!